ord_schema/message_helpers.py

```diff
@@ -338,6 +342,72 @@ def check_compound_identifiers(compound):
         raise ValueError(f'structural identifiers are inconsistent: {smiles}')
 
 
+def get_reaction_smiles(message, validate=True):
+    """Fetches or generates a reaction SMILES.
+
+    Args:
+        message: reaction_pb2.Reaction message.
+        validate: Boolean whether to validate the reaction SMILES with rdkit.
+
+    Returns:
+        Text reaction SMILES.
+
+    Raises:
+        ValueError: If the reaction contains errors.
+    """
+    for identifier in message.identifiers:
+        if identifier.type == reaction_pb2.ReactionIdentifier.REACTION_SMILES:
+            return identifier.value
+    reactants, agents, products = set(), set(), set()
+    roles = reaction_pb2.Compound.ReactionRole()
+    for key in sorted(message.inputs):
+        for compound in message.inputs[key].components:
+            smiles = smiles_from_compound(compound)
+            if compound.reaction_role in [
+                    roles.REAGENT, roles.SOLVENT, roles.CATALYST
+            ]:
+                agents.add(smiles)
+            else:
```

Comment:
This should have an `elif compound.reaction_role == roles.INTERNAL_STANDARD: continue` too

Reply:
Done.